test/csuite/random_abort/main.c

```diff
@@ -266,27 +262,29 @@ thread_run(void *arg)
              * Make sure the modify operation is carried out in an snapshot isolation level with
              * explicit transaction.
              */
-            testutil_check(session->begin_transaction(session, NULL));
-
-            ret = wiredtiger_calc_modify(session, &data, &newv, maxdiff, entries, &nentries);
+            do {
+                testutil_check(session->begin_transaction(session, NULL));
 
-            if (columnar_table)
-                cursor->set_key(cursor, i);
-            else
-                cursor->set_key(cursor, kname);
-
-            if (ret == 0)
-                testutil_check(cursor->modify(cursor, entries, nentries));
-            else {
-                /*
-                 * In case if we couldn't able to generate modify vectors, treat this change as a
-                 * normal update operation.
-                 */
-                cursor->set_value(cursor, &newv);
-                testutil_check(cursor->update(cursor));
-            }
+                if (columnar_table)
+                    cursor->set_key(cursor, i);
+                else
+                    cursor->set_key(cursor, kname);
 
-            testutil_check(session->commit_transaction(session, NULL));
+                ret = wiredtiger_calc_modify(session, &data, &newv, maxdiff, entries, &nentries);
+                if (ret == 0)
+                    ret = cursor->modify(cursor, entries, nentries);
+                else {
+                    /*
+                     * In case if we couldn't able to generate modify vectors, treat this change as
+                     * a normal update operation.
+                     */
+                    cursor->set_value(cursor, &newv);
+                    ret = cursor->update(cursor);
+                }
+                testutil_check(ret = WT_ROLLBACK ? session->commit_transaction(session, NULL) :
```

Comment:
I think you want `ret != WT_ROLLBACK`

Reply:
Also, in the original, we would fail if `cursor->modify()` or `cursor->update()` return any error.  Here on any error other than `WT_ROLLBACK` will commit first then fail after the end of the do loop.  That's probably OK since this is just a test program, but figured I'd double check that this what you intend/expect.

Reply:
> I think you want `ret != WT_ROLLBACK`

Ouch -- thanks -- probably should be `ret == 0`, since could be other errors here.

Reply:
> Also, in the original, we would fail if cursor->modify() or cursor->update() return any error. Here on any error other than WT_ROLLBACK will commit first then fail after the end of the do loop. That's probably OK since this is just a test program, but figured I'd double check that this what you intend/expect.

That's what I'd expect from any program, test or otherwise, I think. If the transaction is selected for rollback, keep trying, but any other error shouldn't happen.